pkg/operator/ceph/object/notification/obc_label_controller.go

```diff
@@ -0,0 +1,221 @@
+}
+
+func addOBCLabelReconciler(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Info("successfully started")
+
+	var cephOBCKind = reflect.TypeOf(bktv1alpha1.ObjectBucketClaim{}).Name()
+	var controllerTypeMeta = metav1.TypeMeta{
+		Kind:       cephOBCKind + "Labels",
+		APIVersion: fmt.Sprintf("%s/%s", cephv1.CustomResourceGroup, cephv1.Version),
+	}
+	// Watch for changes on the OBC CRD object
+	err = c.Watch(&source.Kind{Type: &bktv1alpha1.ObjectBucketClaim{TypeMeta: controllerTypeMeta}}, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a ObjectBucketClaim object and makes changes based on the state read
+// and the ObjectBucketClaim labels
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileOBCLabels) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileOBCLabels) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the ObjectBucketClaim instance
+	cephOBC := &bktv1alpha1.ObjectBucketClaim{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephOBC)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("ObjectBucketClaim %q resource not found. Ignoring since resource must be deleted.", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve ObjectBucketClaim %q", request.NamespacedName)
+	}
+
+	// find the namespace for the ceph cluster (may be different than the namespace of the topic CR)
+	// Make sure a CephCluster is present otherwise do nothing
+	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcileNoName(
+		r.client,
+		r.context,
+		controllerName,
+	)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		if !cephOBC.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		logger.Debug("Ceph cluster not yet present")
+		return reconcileResponse, nil
+	}
+	r.clusterSpec = &cephCluster.Spec
+
+	// DELETE: the CR was deleted
+	if !cephOBC.GetDeletionTimestamp().IsZero() {
```

Comment:
ditto, you can just ignore the DELETE event from the predicate.

Reply:
in the case that the OBC is deleted there is nothing we need to do (ceph would delete all notifications of the bucket automatically).
currently the predicate return "true" only if there was a label change